ivre/db/mongo.py

```diff
@@ -721,6 +721,53 @@ def getscreenshot(port):
         if url == "field":
             return port.get('screendata')
 
+    def set_label(self, flt, group, label, archives=False):
+        """Adds `label` in `group` to every host matching `flt`"""
+        colname = self.colname_oldhosts if archives else self.colname_hosts
+        for host in self.find(colname, flt):
+            labels = host.setdefault('labels', [])
+            try:
+                g_label = (lab for lab in labels if lab['group'] == group).next()
+            except StopIteration:
+                # group did not exist.
+                g_label = {'group': group, 'tags': [label]}
+                labels.append(g_label)
+            else:
+                if label in g_label['tags']:
+                    continue
+                g_label['tags'].append(label)
+            self.db[colname].update_one({"_id": host['_id']},
```

Comment:
Sorry again, missed that in my previous reviews, `.update_one()` and `update_many()` only exist in PyMongo >= 3.0.